var/spack/repos/builtin/packages/mesa/package.py

```diff
@@ -62,7 +63,9 @@ class Mesa(MesonPackage):
     depends_on("flex", type="build")
     depends_on("gettext", type="build")
     # Upperbound on 3.11 because distutils is used for checking py-mako
-    depends_on("python@3:3.11", type="build")
+    depends_on("python@3:3.11", type="build", when="@20:23.3.2")
+    depends_on("python@3.12:", type="build", when="@23.3.3:")
+    depends_on("py-packaging", type="build", when="@23.3.3:")
```

Comment:
According to https://gitlab.freedesktop.org/mesa/mesa/-/blob/main/docs/install.rst, _"Python package `packaging` is required on Python 3.12+"_, so this should probably be:
```suggestion
    depends_on("py-packaging", type="build", when="^python@3.12:")
```